### PR TITLE
404 page, page not found, no internet connection

### PR DESCRIPTION
**Pull request by thasnim-cm**

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -18,62 +19,88 @@ import Sessions from "./views/sessions";
...
+              element={
+                <div className="flex flex-col h-[100vh] items-center justify-center bg-black-100 text-white-500">
+                  <h1 className="text-7xl">404</h1>
+                  <p>Page Not Found.</p>
+                </div>
+              }
+            />
+            <Route
+              path="*"
+              element={<Navigate replace to="/page-not-found" />}
```
Comment (raj-kumar-litmus) [inline src/main.tsx]: Can you change this to `ROUTES.pageNotFound` ?

**Review by raj-kumar-litmus — APPROVED** [review]
Looks good.

**Review by Kajish — APPROVED** [review]

File: src/components/NoInternetConnection.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  if (isOnline) {
+    return props.children;
+  } else {
+    return (
+      <div className="flex h-[100vh] items-center justify-center bg-black-100 text-white-500">
+        No Internet Connection. Please try again later.
+      </div>
+    );
+  }
```
Comment (AdithyaShaj) [inline src/components/NoInternetConnection.tsx]: Keep single return statement and use ternary operation for conditional check

File: src/components/NoInternetConnection.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+export default NoInternetConnection;
```
Comment (AdithyaShaj) [inline src/components/NoInternetConnection.tsx]: Keep the file in the corresponding folder and update the imports

**Review by AdithyaShaj — APPROVED** [review]

File: src/main.tsx [hunk 4]
```diff
@@ -18,62 +19,88 @@ import Sessions from "./views/sessions";
...
+              path={ROUTES.orderReport}
+              element={<App appContent={<OrderReport />} />}
+            />
+            <Route
+              path={ROUTES.connectivityIssues}
+              element={
+                <div className="flex flex-col h-[100vh] items-center justify-center bg-black-100 text-white-500">
+                  <h1 className="text-5xl">Could Not Connect</h1>
+                  <p>
+                    We could not establish connection with the APIs. Please
```
Comment (Kajish) [inline src/main.tsx]: couldnt this be made into a component and imported here instead of this usage

File: src/main.tsx [hunk 5]
```diff
@@ -18,62 +19,88 @@ import Sessions from "./views/sessions";
...
+                  </p>
+                </div>
+              }
+            />
+            <Route
+              path={ROUTES.pageNotFound}
+              element={
+                <div className="flex flex-col h-[100vh] items-center justify-center bg-black-100 text-white-500">
+                  <h1 className="text-7xl">404</h1>
+                  <p>Page Not Found.</p>
```
Comment (Kajish) [inline src/main.tsx]: above comment goes to this too

**Review by Kajish — APPROVED** [review]

**Pull request merged**